lib/thrift/generator/utils.ex

```diff
@@ -60,30 +60,47 @@ defmodule Thrift.Generator.Utils do
   end
 
   @doc """
-  Merge the binaries in an iolist.
+  Optimize an expression that returns an iolist.
```

Comment:
Could you also write a little more about the optimizations that are applied while they're still fresh in your mind?